### DISCUSSION TITLE
Font not kerning

### DISCUSSION THREAD

**Discussion by LazLong** [QuestPDF/QuestPDF#351] [Q&A]
Is there any way to ensure the font kerns appropriately. I'm using Arial and the text is clearly not kerned, it's monospaced. Any thoughts?

Code:
```csharp
var document = Document.Create(container =>
{
    container.Page(page =>
    {
        page.Size(PageSizes.A4);
        page.Margin(12, Unit.Millimetre);
        page.PageColor(Colors.White);
        page.DefaultTextStyle(x=>x.FontSize(9.36f));
        page.Content().Table(table =>
        {
            table.ColumnsDefinition(columns =>
            {
                columns.ConstantColumn(54,Unit.Millimetre);
                columns.ConstantColumn(21,Unit.Millimetre);
                columns.ConstantColumn(193-87,Unit.Millimetre);
            });
            table.Cell().Row(1).Column(1).Image(@"images\dc-logo-long.png");
            table.Cell().Row(1).Column(3).AlignBottom().Text("Level 6").FontFamily(Fonts.Arial).FontSize(21.12f).Bold();
        });
    });
});
````

**Comment by MarcinZiabek** [comment]
Hello 😁 Would you like to share more details? Screenshots?

This is result that I get when using Arial. Kerning (or very similar effect) is clearly visible:

![image](https://user-images.githubusercontent.com/9263853/193239085-649e1d23-8bcf-4af8-b391-8c49b7ba9c62.png)

This is screenshot from Wikipedia:

![image](https://user-images.githubusercontent.com/9263853/193238863-3104833d-de73-4ead-a2b2-bc3b68bac6c7.png)

**Reply by LazLong** [reply]
Apologies, Hello :)

![Screenshot 2022-09-30 133042](https://user-images.githubusercontent.com/8004911/193270225-666f803d-52f3-4ee1-a626-29a3d6b32348.jpg)

Here you go. Monospaced as I mentioned.

**Reply by LazLong** [reply]
Actually, ignore me for the moment. Re-running for that screenshot has the letters more tightly grouped than before. The last L is clearly kerned into the e.

**Reply by MarcinZiabek** [reply]
This is a screenshot from Word (online version). Looks pretty similar to yours:

![image](https://user-images.githubusercontent.com/9263853/193273292-cdfec12d-3de5-489b-8ab0-9f1d082250d4.png)

**Reply by LazLong** [reply]
Okay, I'm copying a document template to fill it in electronically - I can't just use a PDF form for various reasons.

I've opened the original and the generated PDFs and I'm seeing a difference in font size despite Acrobat telling me that both pieces of text are the same font size. The only difference is the line spacing, 1.2 originally, 1.34 from QuestPDF but that shouldn't impact the font size. This is very weird. PostScript is pretty specific in this context normally.

I'll keep digging, something is adrift here and I suspect it may be the original document.

**Reply by MarcinZiabek** [reply]
Text glyphs are rendered using the HarfBuzz library. Font kerning, ligatures, unicode characters, shaping, directionality etc. - all of these features are done by HarfBuzz. There is not much that QuestPDF has control over, and as a result can do incorrectly.

Still, please let me know about your findings. I am open to investigate potential improvements and fixes, if any problem is identified 😁

**Reply by bennetbo** [reply]
Regarding the difference in line spacing:
Are you generating the document on Linux? If so, that sounds similar to [this](https://github.com/QuestPDF/QuestPDF/issues/243#issuecomment-1153420083).
Probably an issue inside SkiaSharp, im afraid there is not much we can do.

**Reply by LazLong** [reply]
Okay, seems the original doc has 21.12pt scaled to 90%, i.e., 19pt - why, oh why would you do that?! Apologies! Loving the library, especially the preview app - perfect tool! Sorry for wasting your time. Have a great weekend!

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
